### PR TITLE
[SW-554] Create and bundle licenses report

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: assembly/allowed-licenses.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {"moduleLicense": "Apache License, Version 2.0"},
+    {"moduleLicense": "Apache Software License, version 1.1"},
+    {"moduleLicense": "The MIT License"},
+    {"moduleLicense": "The 3-Clause BSD License"},
+    {"moduleLicense": "The 2-Clause BSD License"},
+
+    {"moduleLicense": "BSD New license"},
+    {"moduleLicense": "BSD"},
+    {"moduleLicense": "New BSD License"},
+    {"moduleLicense": "BSD New license"},
```
Comment (mn-mikke) [inline assembly/allowed-licenses.json]: It seems to be duplicated with the [line 9](https://github.com/h2oai/sparkling-water/pull/1256/files#diff-e71bc2016e24993261af924f4072d31bR9).

File: assembly/allowed-licenses.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {"moduleLicense": "Public Domain"},
+    {"moduleLicense": "PUBLIC DOMAIN"},
+    {"moduleLicense": "Common Public License Version 1.0"},
+
+    {"moduleLicense": "CDDL + GPLv2 with classpath exception"},
+    {"moduleLicense": "LGPL, version 2.1"},
+    {"moduleLicense": "COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.0"},
+    {"moduleLicense": "GPL2 w/ CPE"},
+    {"moduleLicense": "CDDL 1.1"},
+    {"moduleLicense": "GPL2 w/ CPE"},
```
Comment (mn-mikke) [inline assembly/allowed-licenses.json]: It seems to be duplicated with the [line 23](https://github.com/h2oai/sparkling-water/pull/1256/files#diff-e71bc2016e24993261af924f4072d31bR23).

**Comment by mmalohlava** [comment]
i would propose to publish the result as HTML report in jenkins

**Comment by jakubhava** [comment]
published as part of each jenkins build, for examople -> http://mr-0xc1:8080/job/sparkling_water_pipeline_internal/job/PR-1256/4/artifact/assembly/build/reports/dependency-license/index.html

**Pull request merged**